### ISSUE TITLE
IndexError: Layer [DHCP] not found

### ISSUE THREAD

**Issue by Muz1K1zuM** [csandker/pxethiefy#7]
I´m running pxethiefy connected to the internal network and I´m receiving the following error:
IndexError: Layer [DHCP] not found

<img width="426" alt="image" src="https://github.com/user-attachments/assets/42fc26db-95ff-4b1b-918f-b46e14fa7975" />

Apparently, there is no DHCP traffic. Is this a known issue? or is it related to my particular environment? I can find DHCP traffic with tcpdump:
<img width="443" alt="image" src="https://github.com/user-attachments/assets/3ac75270-e07d-4be1-9ef2-58ec35e5b7cd" />

Thanks in advance.

**Comment by csandker** [comment]
Correct, the error indicates that there is no DHCP packet layer in the captured packet. There could be various reasons for that e.g. that you're sending/listening on the wrong interface or that the packets have not been captured correctly.

If you can reproduce this error reliably it would be a good start to break [before this line](https://github.com/csandker/pxethiefy/blob/main/pxethiefy.py#L278) and check if any captured packets have a DHCP layer. If none are present but you can find valid PXE DHCP responses in wireshark than there is something broken with the packet capture.

**Comment by csandker** [comment]
Ran into the same problem last week. Problem was that some packets did not contain a DHCP layer. Added a longer packet sniffing interval and a check for the DHCP layer. This should be fixed now
